File: String-Algorithms/String-Algorithms-master/Unique String Identification/uniqueString.py
class solution:
    def isUnique(self,input_string):
        mapp = []
        for i in input_string:
            if i not in mapp:
                mapp.append(i)
        return len(mapp) == len(input_string)

input1 = str(input())
sol = solution()
print(sol.isUnique(input1))

File: String-Algorithms/String-Algorithms-master/Unique String Identification/uniqueString.cpp
# include<iostream>
# include<cstdlib>
# include<string>
# include<map>

using namespace std;

class solution
{
public:
    bool isUnique(string input)
    {
        map<char,int> mapp;
        for(int i=0;i<input.size();i++)
        {
            char key = input[i];
            if(mapp.count(key)==0)
            {
                mapp[key] = 1;
            }
            else
            {
                mapp[key]++;
            }
        }
        map<char,int>::iterator it = mapp.begin();
        while(it != mapp.end())
        {
            if(it -> second > 1)
            {
                return false;
            }
            it++;
        }
        return true;
    }
};

int main()
{
    string input1;
    getline(cin,input1);
    solution sol;
    cout << sol.isUnique(input1)<< endl;
    return 0;
}
